apcd-cms/src/apps/extension/views.py

```diff
@@ -8,39 +8,47 @@
 
 logger = logging.getLogger(__name__)
 
+
 class ExtensionFormView(TemplateView):
-    template_name = 'extension_submission_form/extension_submission_form.html'
 
     def dispatch(self, request, *args, **kwargs):
         if not request.user.is_authenticated or not has_apcd_group(request.user):
             return HttpResponseRedirect('/')
         return super(ExtensionFormView, self).dispatch(request, *args, **kwargs)
-   
-    def get_context_data(self, *args, **kwargs):
-            context = super(ExtensionFormView, self).get_context_data(*args, **kwargs)
-
-            user = self.request.user.username
-
-            submitters = apcd_database.get_submitter_for_extend_or_except(user)
-
-            self.request.session['submitters'] = submitters
-
-            def _set_submitter(sub):
-                return {
-                    "submitter_id": sub[0],
-                    "submitter_code": sub[1],
-                    "payor_code": sub[2],
-                    "user_name": sub[3],
-                    "org_name": title_case(sub[4])
-                }
-
-            context['submitters'] = []
-
-            for submitter in submitters: 
-                context['submitters'].append(_set_submitter(submitter))
-
-            return context      
 
+    def get_context_data(self, *args, **kwargs):
+        context = super(ExtensionFormView, self).get_context_data(*args, **kwargs)
+        
+        user = self.request.user.username
+        
+        submitters = apcd_database.get_submitter_for_extend_or_except(user)
+
+        self.request.session['submitters'] = submitters
+
+        def _set_submitter(sub):
+            return {
+                "submitter_id": sub[0],
+                "submitter_code": sub[1],
+                "payor_code": sub[2],
+                "user_name": sub[3],
+                "org_name": title_case(sub[4])
+            }
+        context["submitters"] = []
+
+        for submitter in submitters: 
+            context['submitters'].append(_set_submitter(submitter))
+        return context
+
+    def get_template_names(self):
+        submitters = self.request.session.get("submitters")
+   
+        ## If no submitter_id for user should not show form but show error page
+        if submitters and all((submitter[0] is not None for submitter in submitters)):
+            del self.request.session['submitters']
```

Comment:
These statements are removing the 'submitter' key from the ` request.sessions`. This means that the `post` function breaks and the user is unable to submit an extensions request.

Reply:
Fixed! Thanks